Mathlib/Topology/ContinuousFunction/BoundedCompactlySupported.lean

```diff
@@ -0,0 +1,109 @@
+def compactlySupported (α γ : Type*) [TopologicalSpace α] [NonUnitalNormedRing γ] :
+    TwoSidedIdeal (α →ᵇ γ) :=
+  .mk' {z | HasCompactSupport z} .zero .add .neg' .mul_left .mul_right
+
+variable {α γ : Type*} [TopologicalSpace α] [NonUnitalNormedRing γ]
+
+@[inherit_doc]
+scoped[BoundedContinuousFunction] notation
+  "C_cb(" α ", " γ ")" => compactlySupported α γ
+
+lemma mem_compactlySupported {f : α →ᵇ γ} :
+    f ∈ C_cb(α, γ) ↔ HasCompactSupport f :=
+  TwoSidedIdeal.mem_mk' {z : α →ᵇ γ | HasCompactSupport z} .zero .add .neg' .mul_left .mul_right f
+
+lemma exist_norm_eq [c : Nonempty α] {f : α →ᵇ γ} (h : f ∈ C_cb(α, γ)) : ∃ (x : α),
+    ‖f x‖ = ‖f‖ := by
+  by_cases hs : (tsupport f).Nonempty
+  · obtain ⟨x, _, hmax⟩ := mem_compactlySupported.mp h |>.exists_isMaxOn hs <|
+      (map_continuous f).norm.continuousOn
+    refine ⟨x, le_antisymm (norm_coe_le_norm f x) (norm_le (norm_nonneg _) |>.mpr fun y ↦ ?_)⟩
+    by_cases hy : y ∈ tsupport f
+    · exact hmax hy
+    · simp [image_eq_zero_of_nmem_tsupport hy]
+  · suffices f = 0 by simp [this]
+    rwa [not_nonempty_iff_eq_empty, tsupport_eq_empty_iff, ← coe_zero, ← DFunLike.ext'_iff] at hs
+
+theorem norm_lt_iff_of_compactlySupported {f : α →ᵇ γ} (h : f ∈ C_cb(α, γ)) {M : ℝ}
+    (M0 : 0 < M) : ‖f‖ < M ↔ ∀ (x : α), ‖f x‖ < M := by
+  refine ⟨fun hn x ↦ lt_of_le_of_lt (norm_coe_le_norm f x) hn, ?_⟩
+  · obtain (he | he) := isEmpty_or_nonempty α
+    · simpa
+    · obtain ⟨x, hx⟩ := exist_norm_eq h
+      exact fun h ↦ hx ▸ h x
+
+theorem norm_lt_iff_of_nonempty_compactlySupported [c : Nonempty α] {f : α →ᵇ γ}
+    (h : f ∈ C_cb(α, γ)) {M : ℝ} : ‖f‖ < M ↔ ∀ (x : α), ‖f x‖ < M := by
+  obtain (hM | hM) := lt_or_le 0 M
+  · exact norm_lt_iff_of_compactlySupported h hM
+  · exact ⟨fun h ↦ False.elim <| (h.trans_le hM).not_le (by positivity),
+      fun h ↦ False.elim <| (h (Classical.arbitrary α) |>.trans_le hM).not_le (by positivity)⟩
+
+theorem compactlySupported_eq_top_of_isCompact (h : IsCompact (Set.univ : Set α)) :
+    C_cb(α, γ) = ⊤ :=
+  eq_top_iff.mpr fun _ _ ↦ h.of_isClosed_subset (isClosed_tsupport _) (subset_univ _)
+
+/- This is intentionally not marked `@[simp]` to prevent Lean looking for a `CompactSpace α`
+instance every time it sees `C_cb(α, γ)`. -/
+theorem compactlySupported_eq_top [CompactSpace α] : C_cb(α, γ) = ⊤ :=
+  compactlySupported_eq_top_of_isCompact CompactSpace.isCompact_univ
+
+theorem compactlySupported_eq_top_iff [Nontrivial γ] :
+    C_cb(α, γ) = ⊤ ↔ IsCompact (Set.univ : Set α) := by
+  refine ⟨fun h ↦ ?_, compactlySupported_eq_top_of_isCompact⟩
+  obtain ⟨x, hx⟩ := exists_ne (0 : γ)
+  simpa [tsupport, Function.support_const hx]
+    using (mem_compactlySupported (f := const α x).mp (by simp [h])).isCompact
+
+lemma hasCompactSupport_mul_of_continuous_compactlySupported (f : α →ᵇ γ) (hf : f ∈ C_cb(α, γ))
+    (g : α → γ) : HasCompactSupport ((g * f : α → γ)) := HasCompactSupport.mul_left
+  (mem_compactlySupported.mp hf)
+
```

Comment:
I don't think this lemma is adding much value, and there's no real reason to require `f ∈ C_cb(α, γ)` over `HasCompactSupport f` (or `HasCompactSupport g`, for that matter). You can just inline the proof below where you needed it.
```suggestion
```